### PR TITLE
Add WASP-10b end-to-end integration test

### PR DESCRIPTION
**Pull request by vterron**
Reduce a transit of exoplanet [WASP-10b](https://en.wikipedia.org/wiki/WASP-10b) and compare the resulting light curve to that in a [golden file](https://softwareengineering.stackexchange.com/questions/358786/what-is-golden-files).

### CODE REVIEW

File: database.py [hunk 1]
```diff
@@ -359,10 +372,25 @@ def __init__(self, path, dtype = numpy.longdouble):
...
 
-    def __del__(self):
+    def _close(self):
         self._cursor.close()
         self.connection.close()
 
+    # TODO(vterron): migrate all uses of LEMONdB to the 'with' statement.
+    def __del__(self):
+        try:
+            self._close()
```
Comment (pablogsal) [inline database.py:382]: 3hxef2

Comment (vterron) [reply database.py:382]: :eyes:

File: test/test_database.py [hunk 2]
```diff
@@ -2394,35 +2390,32 @@ def make_db(name, value):
         for name, values in metaproperties.iteritems():
 
             # AttributeError is raised if the property is not set
-            db1 = LEMONdB(":memory:")
-            with self.assertRaisesRegexp(AttributeError, regexp % name):
-                getattr(db1, name)
+            with LEMONdB(":memory:") as db:
```
Comment (vterron) [inline test/test_database.py:2393]: These variables got accidentally renamed; fixing it in a follow-up CL.

File: database.py [hunk 3]
```diff
@@ -1561,7 +1561,7 @@ def field_name(self):
 
         def startswith_counter(prefix, names):
             """ Return the number of strings in 'names' starting with 'prefix' """
-            return len(x for x in names if x.startswith(prefix))
```
Comment (vterron) [inline database.py]: Thanks for fixing this. For future reference, this results in error "object of type 'generator' has no len()" I'm surprised this went unnoticed for so many years.

**Pull request merged**